projects/storefrontlib/src/lib/store-finder/components/store-finder-grid/store-finder-grid.component.ts

```diff
@@ -39,17 +34,12 @@ export class StoreFinderGridComponent implements OnInit {
       }
     }
 
-    this.store
-      .pipe(select(fromStore.getFindStoresEntities))
-      .subscribe(locations => {
-        if (
-          locations.pointOfServices &&
-          locations.pointOfServices.length === 1
-        ) {
-          this.viewStore(locations.pointOfServices[0]);
-        }
-        this.locations = locations;
-      });
+    this.storeFinderService.getFindStoresEntities().subscribe(locations => {
```

Comment:
`locations` can be also `Observable`.
```
this.locations$ = this.storeFinderService.getFindStoresEntities().pipe(
      tap(locations => {
        if (
          locations.pointOfServices &&
          locations.pointOfServices.length === 1
        ) {
          this.viewStore(locations.pointOfServices[0]);
        }
      })
    );
```
If you don't want to change the codes, you need to `unsubscribe` the subscripton in `ngOnDestroy`. Otherwise, there will be a leak.